sql/5_0_2-to-5_0_3_upgrade.sql

```diff
@@ -515,3 +515,82 @@ ALTER TABLE `api_token` ADD `token_api` varchar(40);
 #IfMissingColumn api_token patient_id
 ALTER TABLE `api_token` ADD `patient_id` bigint(20) NOT NULL;
 #EndIf
+
+#IfNotTable fhir_forced_id
+CREATE TABLE `fhir_forced_id` (
+    `pid` bigint(20) NOT NULL,
+    `forced_id` varchar(100) NOT NULL,
+    `resource_pid` bigint(20) NOT NULL,
+    `resource_type` varchar(100) DEFAULT '',
+    PRIMARY KEY (`pid`),
+    UNIQUE KEY `idx_forcedid_resid` (`resource_pid`),
+    UNIQUE KEY `idx_forcedid_type_resid` (`resource_type`,`resource_pid`),
+    KEY `idx_forcedid_type_forcedid` (`resource_type`,`forced_id`)
+) ENGINE=InnoDB;
+#EndIf
+
+#IfNotTable fhir_resource
+CREATE TABLE `fhir_resource` (
+    `res_id` bigint(20) NOT NULL,
+    `res_deleted_at` datetime DEFAULT NULL,
+    `res_version` varchar(7) DEFAULT NULL,
+    `has_tags` bit(1) NOT NULL,
+    `res_published` datetime NOT NULL,
+    `res_updated` datetime NOT NULL,
+    `sp_has_links` bit(1) DEFAULT NULL,
+    `hash_sha256` varchar(64) DEFAULT NULL,
+    `sp_index_status` bigint(20) DEFAULT NULL,
+    `res_language` varchar(20) DEFAULT NULL,
+    `sp_cmpstr_uniq_present` bit(1) DEFAULT NULL,
+    `sp_coords_present` bit(1) DEFAULT NULL,
+    `sp_date_present` bit(1) DEFAULT NULL,
+    `sp_number_present` bit(1) DEFAULT NULL,
+    `sp_quantity_present` bit(1) DEFAULT NULL,
+    `sp_string_present` bit(1) DEFAULT NULL,
+    `sp_token_present` bit(1) DEFAULT NULL,
+    `sp_uri_present` bit(1) DEFAULT NULL,
+    `res_profile` varchar(200) DEFAULT NULL,
+    `res_type` varchar(30) DEFAULT NULL,
+    `res_ver` bigint(20) DEFAULT NULL,
+    `forced_id_pid` bigint(20) DEFAULT NULL,
+    PRIMARY KEY (`res_id`),
+    KEY `idx_res_date` (`res_updated`),
+    KEY `idx_res_lang` (`res_type`,`res_language`),
+    KEY `idx_res_profile` (`res_profile`),
+    KEY `idx_res_type` (`res_type`),
+    KEY `idx_indexstatus` (`sp_index_status`),
+    KEY `fk_resource_forcedid` (`forced_id_pid`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+#EndIf
+
+#IfNotTable fhir_res_ver
+CREATE TABLE `fhir_res_ver` (
+    `pid` bigint(20) NOT NULL,
+    `res_deleted_at` datetime DEFAULT NULL,
+    `res_version` varchar(7) DEFAULT NULL,
+    `has_tags` bit(1) NOT NULL,
+    `res_published` datetime NOT NULL,
+    `res_updated` datetime NOT NULL,
```

Comment:
need to have datetime default to NULL (if nothing is given on the insert)

Reply:
I went ahead and changed all the datetimes in default and upgrade. I'm assuming this is okay.
@bradymiller

Reply:
@sjpadgett , I would only focus on the items you are adding. There is an ongoing project looking at other items since the ones that already exist may have to also convert underlying data.

Reply:
However, if possible, keep the ones in this upgrade script and companion database.sql changes since those haven't had a chance to cause problems yet :)

Reply:
huh, i'd expect if it's a problem for one then a potential problem for all. Still, I get it and will revert.

Reply:
On further thought, should also not change the other in upgrade 5.0.2->5.0.3 since those have been there long enough where we need to change the DEFAULTS also in the upgrade scripts via #IfNotColumnTypeDefault calls . These changes in previous stuff will require something like (per stuff @ShubhamBansal1997 is working on):
1. modify in database.sql
2. use #IfNotColumnTypeDefault in upgrade script
3. then clean up any illegal entries in the data

Reply:
Okay, i'm reverting now. I'm just glad it's not me dealing with this issue:)